hub/api/tests/test_api.py

```diff
@@ -260,15 +260,27 @@ def test_shape_property(memory_ds):
 def test_dtype(memory_ds: Dataset):
     tensor = memory_ds.create_tensor("tensor")
     dtyped_tensor = memory_ds.create_tensor("dtyped_tensor", dtype="uint8")
-
-    assert tensor.meta.dtype == None
-    assert dtyped_tensor.meta.dtype == "uint8"
+    np_dtyped_tensor = memory_ds.create_tensor("np_dtyped_tensor", dtype=np.float)
+    py_dtyped_tensor = memory_ds.create_tensor("py_dtyped_tensor", dtype=float)
+
```

Comment:
add test case to make sure that tensor.meta.dtype is always either None or of type string